### PR TITLE
Instrument address space passes and friends

### PR DESCRIPTION
**Pull request by pchintalapudi**

### CODE REVIEW

**Review by vchuravy — APPROVED** [review]

**Comment by giordano** [comment]
Tests are failing on all systems here.

File: src/llvm-remove-addrspaces.cpp [hunk 1]
```diff
@@ -452,6 +453,7 @@ bool removeAddrspaces(Module &M, AddrspaceRemapFunction ASRemapper)
         }
     }
 
+    verifyModule(M);
```
Comment (vtjnash) [inline src/llvm-remove-addrspaces.cpp:456]:
```suggestion
```

File: src/llvm-remove-ni.cpp [hunk 2]
```diff
@@ -32,6 +33,7 @@ static bool removeNI(Module &M)
     }
     dlstr.erase(nistart, niend - nistart);
     M.setDataLayout(dlstr);
+    assert(!verifyModule(M));
```
Comment (vtjnash) [inline src/llvm-remove-ni.cpp:36]:
```suggestion
```

**Comment by vtjnash** [comment]
`verifyModule` is a pass, and shouldn't be included as part of other passes (it is also very expensive)

**Comment by pchintalapudi** [comment]
Superseded by #46407 and #46387

**Pull request closed without merging**